src/pages/Entities.vue

```diff
@@ -0,0 +1,247 @@
+  'count-mentions', '-count-mentions'
+];
+const OrderByDefault = '-count';
+
+export default {
+  data: () => ({
+    TabObservingList,
+    TabBrowseList,
+    paginationPerPage: 20,
+    paginationCurrentPage: 1,
+    paginationTotalRows: 0,
+    items: [],
+    isLoading: false,
+  }),
+  computed: {
+    paginationList() {
+      return {
+        perPage: this.paginationPerPage,
+        currentPage: this.paginationCurrentPage,
+        totalRows: this.paginationTotalRows,
+      };
+    },
+    observedItems: {
+      get() {
+        return (this.$route.query.items || '').split(',').filter(d => !!d);
+      },
+      set(items) {
+        this.$navigation.updateQueryParametersWithHistory({
+          items: items.join(','),
+        });
+      },
+    },
+    tab: {
+      get() {
+        return [
+          TabObservingList,
+          TabBrowseList,
+        ].includes(this.$route.query.tab)
+          ? this.$route.query.tab
+          : TabBrowseList;
+      },
+      set(tab) {
+        this.$navigation.updateQueryParameters({
+          tab,
+        });
+      },
+    },
+    serviceQuery: {
+      get() {
+        return {
+          q: this.suggestionQuery,
+          limit: this.paginationPerPage,
+          page: this.paginationCurrentPage,
+          orderBy: this.orderBy,
+        };
+      },
+    },
+    selectedId() {
+      return this.$route.params.entity_id;
+    },
+    orderByOptions() {
+      return OrderByOptions.map(value => ({
+        value,
+        text: this.$t(`sort_${value}`),
+      }))
+    },
+    orderBy: {
+      get() {
+        return OrderByOptions.includes(this.$route.query.orderBy)
+          ? this.$route.query.orderBy
+          : OrderByDefault;
+      },
+      set(orderBy) {
+        this.$navigation.updateQueryParametersWithHistory({
+          orderBy,
+        });
+      },
+    },
+    suggestionQuery: {
+      get() {
+        return this.$route.query.q ?? '';
+      },
+      set(q) {
+        this.paginationCurrentPage = 1;
+        this.$navigation.updateQueryParametersWithHistory({
+          q,
+        });
+      },
+    },
+  },
+  methods: {
+    switchTab(tab) {
+      this.$navigation.updateQueryParametersWithHistory({
+        tab,
+      });
+    },
+    handleToggleObserved(item) {
+      if (this.observedItems.includes(item.uid)) {
+        this.observedItems = this.observedItems.filter(uid => uid !== item.uid);
+      } else {
+        this.observedItems = this.observedItems.concat(item.uid);
+      }
+    },
+    resetObservedItems() {
+      this.observedItems = [];
+    },
+    changePage(page = 1) {
+      this.paginationCurrentPage = page;
+    },
+  },
+  watch: {
+    serviceQuery: {
+      async handler(params, oldParams) {
+        const newParamsStr = JSON.stringify(params)
+        const oldParamsStr = JSON.stringify(oldParams)
+        if (newParamsStr === oldParamsStr) {
+          console.warn(`Params are the same: ${newParamsStr} ${oldParamsStr}`)
```

Comment:
As per our discussion the other day - if there is time it would be great to check why watcher is triggered when parameters are not changed. `deep: true` should not allow this to happen. But in any case let's remove `console.warn`?